src/zenml/stack_stores/sql_stack_store.py

```diff
@@ -273,27 +333,44 @@ def deregister_stack(self, name: str) -> None:
 
     # Private interface implementations:
 
-    def _create_stack(
-        self, name: str, stack_configuration: Dict[StackComponentType, str]
+    def _save_stack(
+        self,
+        name: str,
+        stack_configuration: Dict[StackComponentType, str],
     ) -> None:
-        """Add a stack to storage.
+        """Save a stack.
 
         Args:
             name: The name to save the stack as.
             stack_configuration: Dict[StackComponentType, str] to persist.
         """
         with Session(self.engine) as session:
-            stack = ZenStack(name=name, created_by=1)
-            session.add(stack)
+            stack = session.exec(
+                select(ZenStack).where(ZenStack.name == name)
+            ).first()
```

Comment:
Similar to the local store, I think we should handle a stack with that name already existing

Reply:
Yes I think this is now handled.